### PR TITLE
Updated JavaScript Solutions

### PR DESCRIPTION
**Pull request by YourMobileGeek**
#### Provided solutions for Chapter 01: 1.2 & 1.9 and Chapter 02: 2.2 in the JavaScript folder

### CODE REVIEW

File: JavaScript/chapter02/2.2 - Return Kth to Last/solution.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        console.log(head.value);
+        findKthToLast(k, head.next)
+    } else {
+        findKthToLast( k-1, head.next);
+    }
+};
+
+/* Tests */ 
+var a = new linkedList('1');
+var a = new linkedList('1');
```
[the remark this answers is not in the document]
Comment (YourMobileGeek) [reply JavaScript/chapter02/2.2 - Return Kth to Last/solution.js]: No particular reason actually, just realized it was repeated just now when you pointed it out 😅

File: JavaScript/chapter01/1.9 - String Rotation/solution.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+/* Assume you have a method isSubstring which checks if one word is a substring
+of another. Given two strings, sl and s2, write code to check if s2 is a rotation of sl using only one
+call to isSubstring (e.g., "waterbottle" is a rotation of"erbottlewat") */
+
+var StringRotate = function(string1, string2) {
+    if (string1.length !== string2.length ){
+        return false; 
+    }
+    return ( string2 + string1 ).includes(string1); // one call of Substring
```
[the remark this answers is not in the document]
Comment (YourMobileGeek) [reply JavaScript/chapter01/1.9 - String Rotation/solution.js]: Thanks! To answer your question too I would consider ES6, this was just one of the first solutions that came to mind.

File: JavaScript/chapter02/2.2 - Return Kth to Last/solution.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (fvntr) [reply JavaScript/chapter02/2.2 - Return Kth to Last/solution.js]: can you delete this line, then we can merge it in

[the remark this answers is not in the document]
Comment (YourMobileGeek) [reply JavaScript/chapter02/2.2 - Return Kth to Last/solution.js]: Sounds good, I'll go ahead and work on that now

**Pull request merged**